json/test_boost_json.cpp

```diff
@@ -0,0 +1,74 @@
+#include <boost/json.hpp>
+#include <boost/json/src.hpp>
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <libnotify.hpp>
+#include <unistd.h>
+
+using namespace std;
+
+struct coordinate_t {
+  float x;
+  float y;
+  float z;
+
+  auto operator<=>(const coordinate_t&) const = default;
+
+  friend ostream& operator<< (ostream &out, const coordinate_t &point) {
+    out << "coordinate_t {x: " << point.x
+        << ", y: " << point.y
+        << ", z: " << point.z << "}";
+    return out;
+  }
+};
+
+string read_file(const string& filename) {
+  ifstream f(filename);
+  if (!f) {
+    return {};
+  }
+  return string(istreambuf_iterator<char>(f),
+                istreambuf_iterator<char>());
+}
+
+coordinate_t calc(const string& text) {
+  auto x = 0.0, y = 0.0, z = 0.0;
+  auto len = 0;
+
+  auto jv = boost::json::parse(text);
```

Comment:
this uses the default memory resource, which is the slowest

Reply:
What's the best approach to use here? Please note that all tests assume that JSON file is already loaded into memory, and they parse the whole blob. There could be optimizations related to streaming operations, but it wouldn't be fair for implementations not supporting it (it's possible that streaming operations will be addressed in a new test suite in the future though).

Also, I'd like to note that this project doesn't assume micro-optimizations including using custom memory allocators like jemalloc, or relying on tribal knowledge not covered in tutorials. I've followed the official documentation, and if it's something not covered there, they need to update the documentation first, and after that, we'll update the tests. However, I have to admit that I could miss something, as Boost.JSON docs are little bit challenging to follow.

Reply:
Seems ok then